Cython/Compiler/Parsing.py

```diff
@@ -1542,10 +1544,15 @@ def p_expression_or_assignment(s):
         return Nodes.ExprStatNode(expr.pos, expr=expr)
 
     rhs = expr_list[-1]
+    if rhs is None:
+        assert s.fault_tolerant  # i.e.: This should never happen unless we're in fault-tolerant mode.
+        pos = s.position()
+    else:
+        pos = rhs.pos
```

Comment:
I think this should best point to the position of the last `=` sign, rather than the RHS node. That would make the error reporting more accurate and solve the problem at hand at the same time.